packages/components/src/range-control/index.js

```diff
@@ -30,6 +30,7 @@ function RangeControl( {
 	min,
 	max,
 	setState,
+	required = true,
```

Comment:
Can you explain how this is related to the changes, and why we're suddenly setting `required` as the default?

Reply:
Hi @aduth, required ensures that an empty value is treated as false by checkValidity. We are not changing the behavior, previously by default empty values were already treated as invalid:
```
const newValue = event.target.value;
const newNumericValue = parseInt( newValue, 10 );
isNaN( newNumericValue ) === true
```

Reply:
> required ensures that an empty value is treated as false by checkValidity. We are not changing the behavior, previously by default empty values were already treated as invalid:

I suppose that's true, though it seems like something which ought to be opted in by the renderer. It feels a bit strange that if I rendered `<RangeControl />`, the element is immediately considered in a state of being invalid in whichever form its placed. I'm failing to recall why we chose to implement it the way we did, but it seems like an empty value should either be considered explicitly as an additional condition with `! checkValidity || value === ''`, or signalled immediately as a valid value equivalent to `onChange( { value: undefined } )`. I think this latter one is what we chose to avoid previously because it made some blocks need to be aware to reset their default attributes?

Reply:
> I suppose that's true, though it seems like something which ought to be opted in by the renderer. It feels a bit strange that if I rendered <RangeControl />, the element is immediately considered in a state of being invalid in whichever form its placed. 

Hi @aduth, the situation is equivalent to any case where we have a required input field.
The following field is also invalid by default until the user types something.
e.g:
```
<input type="text" name="usrname" required>
```
It seems invalid fields at the start before the user types something are a normal and expected situation.
Given the way the input field is used inside the RangeControl it also seems fine that <RangeControl /> is invalid by default. We give the option to the user to set required to false and make the component valid. But in all cases where we use RangeControl, we want a value so if we don't pass one we are in an invalid state. In most cases, the value comes from an attribute that has a default value so we will not have the range control in an invalid state.

> ! checkValidity || value === ''

We can follow this approach, do you think it would make sense to still have a flag that allows undefined/empty values?

> or signalled immediately as a valid value equivalent to onChange( { value: undefined } ). I think this latter one is what we chose to avoid previously because it made some blocks need to be aware to reset their default attributes?

Exactly, most of the times the value comes from an attribute with a default value, and setting the attribute to undefined would not reset to the default value making the blocks not work as expected. But if we set the default values the result would also not be great. Imagine we value a RangeControl with current value 1, the attribute has a default value of 2, and the user wants to set the value to 3. If we called onChange( { value: undefined } ) as soon as the form was empty, and the default attribute was set when the attribute was undefined, when the user deleted the 1 to type 3, a value of 2 would immediately appear.

Reply:
>The following field is also invalid by default until the user types something.
>e.g:
>
>```
><input type="text" name="usrname" required>
>```
>It seems invalid fields at the start before the user types something are a normal and expected situation.

>>! checkValidity || value === ''
>
>We can follow this approach, do you think it would make sense to still have a flag that allows >undefined/empty values?

To me, the main thing is that `required` be opt-in. In your first example, you had to literally write `required` for that behavior to take effect. What we're talking about in the current state of this pull request is that behavior taking effect by the default rendering of a `<RangeControl />`.

To your second point, I guess it depends if we can make it work where the control would signal `undefined` when the field is empty, and in order to get around the previous issues we had with block implementations, those usages should be updated to use `required` (in which case, `undefined` would not be passed as the value).

Would that work? Is it sensible?

Reply:
Hi @aduth I updated the code to follow your suggestion, required is now opt-in and in our own usages, we use this option.